Modules/_io/textio.c

```diff
@@ -248,9 +248,11 @@ _io_IncrementalNewlineDecoder___init___impl(nldecoder_object *self,
                                             PyObject *errors)
 /*[clinic end generated code: output=fbd04d443e764ec2 input=89db6b19c6b126bf]*/
 {
+    Py_XDECREF(self->decoder);
```

Comment:
There is a race condition, destructor of `self->decoder` can call `IncrementalNewlineDecoder.__init__` . Use Py_XSETREF().